cpp/references/data_structures/typedefs.hpp

```diff
@@ -0,0 +1,18 @@
+
```

Comment:
Reconsidering this file. The way its written here, we are doing a bunch of unrelated includes together. If you wanted to include Map, you'd have to also include Set and Vector, even if you didn't need them.

Why don't we instead do map.h, set.h, vector.h, (just put the list_digraph inclusion and all the typedefs in list_digraph.h) and the #include <type> #define Type std::type combo in each file.